allennlp_models/rc/dataset_readers/transformer_squad.py

```diff
@@ -20,23 +20,39 @@
 @DatasetReader.register("transformer_squad")
 class TransformerSquadReader(DatasetReader):
     """
-    Reads a JSON-formatted SQuAD file and returns a ``Dataset`` where the ``Instances`` have four
-    fields:
-     * ``question_with_context``, a ``TextField`` that contains the concatenation of question and context,
-     * ``answer_span``, a ``SpanField`` into the ``question`` ``TextField`` denoting the answer.
-     * ``context_span`` a ``SpanField`` into the ``question`` ``TextField`` denoting the context, i.e., the part of
+    !!! Note
+        If you're training on SQuAD v1.1 you should use the [`squad1()`](#squad1) classmethod
+        to instantiate this reader, and for SQuAD v2.0 you should use the [`squad2()`](#squad2).
+
+    Dataset reader suitable for JSON-formatted SQuAD-like datasets to be used with a transformer-based
+    QA model, such as [`TransformerQA`](../../models/transformer_qa#TransformerQA).
+
+    It will generate `Instances` with the following fields:
+
+     * `question_with_context`, a `TextField` that contains the concatenation of question and context,
+     * `answer_span`, a `SpanField` into the `question` `TextField` denoting the answer.
+     * `context_span`, a `SpanField` into the `question` `TextField` denoting the context, i.e., the part of
        the text that potential answers can come from.
-     * A ``MetadataField`` that stores the instance's ID, the original question, the original passage text, both of
-       these in tokenized form, and the gold answer strings, accessible as ``metadata['id']``,
-       ``metadata['question']``, ``metadata['context']``, ``metadata['question_tokens']``,
-       ``metadata['context_tokens']``, and ``metadata['answers']. This is so that we can more easily use the
+     * `metadata`, a `MetadataField` that stores the instance's ID, the original question, the original
+       passage text, both of these in tokenized form, and the gold answer strings, accessible as
+       `metadata['id']`, `metadata['question']`, `metadata['context']`, `metadata['question_tokens']`,
+       `metadata['context_tokens']`, and `metadata['answers']`. This is so that we can more easily use the
        official SQuAD evaluation script to get metrics.
+     * `cls_index`, a `IndexField` that holds the index of the `[CLS]` token within the
+       `question_with_context` field. This is only present if `use_cls_token_for_unanswerable`
+       is set to `True`.
```

Comment:
Can't we just find that token? Do we have to specify it in the instance?

Reply:
I don't know of a reliable way to find it

Reply:
You can read it out of the huggingface tokenizers. They have calls for that.